include/rosidl_dynamic_typesupport/api/serialization_support_interface.h

```diff
@@ -1318,26 +1318,68 @@ struct rosidl_dynamic_typesupport_serialization_support_interface_s
 
 
   // DYNAMIC DATA NESTED
+
+  rcutils_ret_t (* dynamic_data_get_nested_data)(
+    rosidl_dynamic_typesupport_serialization_support_impl_t * serialization_support,
+    const rosidl_dynamic_typesupport_dynamic_data_impl_t * dynamic_data,
+    rosidl_dynamic_typesupport_member_id_t id,
+    rcutils_allocator_t * allocator,
+    rosidl_dynamic_typesupport_dynamic_data_impl_t * value);  // OUT (copies)
+
+  rcutils_ret_t (* dynamic_data_set_nested_data)(
+    rosidl_dynamic_typesupport_serialization_support_impl_t * serialization_support,
+    rosidl_dynamic_typesupport_dynamic_data_impl_t * dynamic_data,
+    rosidl_dynamic_typesupport_member_id_t id,
+    rosidl_dynamic_typesupport_dynamic_data_impl_t * value);
+
+  // This deep copies the passed data
+  rcutils_ret_t (* dynamic_data_insert_nested_data_copy)(
+    rosidl_dynamic_typesupport_serialization_support_impl_t * serialization_support,
+    rosidl_dynamic_typesupport_dynamic_data_impl_t * dynamic_data,
+    const rosidl_dynamic_typesupport_dynamic_data_impl_t * value,
+    rosidl_dynamic_typesupport_member_id_t * out_id);  // OUT
+
+  rcutils_ret_t (* dynamic_data_insert_nested_data)(
+    rosidl_dynamic_typesupport_serialization_support_impl_t * serialization_support,
+    rosidl_dynamic_typesupport_dynamic_data_impl_t * dynamic_data,
+    rosidl_dynamic_typesupport_dynamic_data_impl_t * value,
+    rosidl_dynamic_typesupport_member_id_t * out_id);  // OUT
+
+  /**
+   * \deprecated `dynamic_data_get_complex_value` is deprecated in favor of
+   *   `dynamic_data_get_nested_data`
+   */
```

Comment:
Could we just rename the fields in the struct?

I am assuming that the API of this package is the one in `dynamic_data.h`, and that the struct defined in this header is the interface that the implementations should provide.

Reply:
If we rename it, `rosidl_dynamic_typesupport_fastrps` (and any other dependent packages) will be broken until the other PR is merged.

Reply:
We need an opinion from the project lead. @clalancette what do you think?